examples/huggingface-transformers/server.py

```diff
@@ -0,0 +1,197 @@
+using the DeepSparse Engine as the inference backend
+
+##########
+Command help:
+usage: server.py [-h] [-c NUM_CORES] [-a ADDRESS] [-p PORT] onnx_filepath
+
+Host a BERT ONNX model as a server, using the DeepSparse Engine and Flask
+
+positional arguments:
+  onnx_filepath         The full filepath of the ONNX model file or SparseZoo
+                        stub to the model
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the analysis on,
+                        defaults to all physical cores available on the system
+  -a ADDRESS, --address ADDRESS
+                        The IP address of the hosted model
+  -p PORT, --port PORT  The port that the model is hosted on
+
+##########
+Example command for running using a model from sparsezoo:
+python server.py \
+    zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned-moderate
+"""
+import argparse
+import json
+import time
+from typing import Any, Callable, List
+
+import flask
+from flask import Flask, jsonify, make_response
+from flask_cors import CORS
+
+from deepsparse.transformers import pipeline
+
+
+app = Flask(__name__)
+CORS(app)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description=(
+            "Host a BERT ONNX model as a server, using the DeepSparse Engine and Flask"
+        )
+    )
+    parser.add_argument(
+        "onnx_filepath",
+        type=str,
+        help="The full filepath of the ONNX model file or SparseZoo stub to the model",
+    )
+
+    parser.add_argument(
+        "-c",
+        "--num-cores",
+        type=int,
+        default=0,
+        help=(
+            "The number of physical cores to run the analysis on, "
+            "defaults to all physical cores available on the system"
+        ),
+    )
+    parser.add_argument(
+        "-a",
+        "--address",
+        type=str,
+        default="0.0.0.0",
+        help="The IP address of the hosted model",
+    )
+    parser.add_argument(
+        "-p",
+        "--port",
+        type=str,
+        default="5543",
+        help="The port that the model is hosted on",
+    )
+    return parser.parse_args()
+
+
+def run_server(
+    predictor: Callable[[Any, Any], Any],
+    host: str = "0.0.0.0",
+    port: str = "5543",
+    info: str = None,
+):
+    """
+    Method to create routes and serve predictor on the specified port
+    """
+    assert callable(predictor), "predictor should be callable"
+
+    @app.route("/predict", methods=["POST"])
+    def predict():
+        """
+        Expects data in json format with question and context
+
+        '{"question":"what's my name?","context":"my name is snorlax"}'
+
+        :returns: A json with results from predictor and the inputs
+        """
+        data = json.loads(flask.request.get_data())
+
+        assert "question" in data and "context" in data
```

Comment:
we probably also want to run an assert that no values are < -1 or greater than vocab size

Reply:
scratch that. Tokenization is not there. given we ware tying to have a fair benchmark it probably makes sense to have a fully processed and tokenized benchmark happening before we do any type of inference. Logically tokenization can happen on client because that is non GPU bound and only focusing on inference on input tokens will give a clearer representation of gains. I don't have data top back this up but by pushing the tokenization to the dataset this should make the gains found by deepsparse clearer.